ParseLiveQuery/src/main/java/com/parse/OkHttp3SocketClientFactory.java

```diff
@@ -0,0 +1,127 @@
+package com.parse;
+
+import android.util.Log;
+
+import java.net.URI;
+import java.util.Locale;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.WebSocket;
+import okhttp3.WebSocketListener;
+import okio.ByteString;
+
+/* package */
+
+/* package */ public class OkHttp3SocketClientFactory implements WebSocketClientFactory {
+
+    OkHttpClient mClient;
+
+    public OkHttp3SocketClientFactory(OkHttpClient client) {
+        mClient = client;
+    }
+
+    public OkHttp3SocketClientFactory() {
+        mClient = new OkHttpClient();
+    }
+
+    @Override
+    public WebSocketClient createInstance(WebSocketClient.WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+        return new OkHttp3WebSocketClient(mClient, webSocketClientCallback, hostUrl);
+    }
+
+    public WebSocketClient createInstance(OkHttpClient client, WebSocketClient.WebSocketClientCallback webSocketClientCallback, URI hostUrl) {
+        return new OkHttp3WebSocketClient(mClient, webSocketClientCallback, hostUrl);
+    }
+
+    class OkHttp3WebSocketClient implements WebSocketClient {
```

Comment:
TubeSocketClient has it reversed (the factory is inside TubeSocketClient).  That works for static classes, but this class requires an instance to OkHttp3 so Java complains.   I reversed it because it's actually the factory class we need access to...the constructors in OkHttp3WebSocketCient/TubeSocketClient are private methods and I don't think the user needs access to that stuff in general so keeping it in the same file avoids having to expose those constructors as methods.